src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -0,0 +1,1497 @@
+        // the IIS and IIS Express bindings. 
+        private IISSettings _currentIISSettings;
+        private IProjectThreadingService _projectThreadingService;
+        private IProjectThreadingService ProjectThreadingService
+        {
+            get
+            {
+                if (_projectThreadingService == null)
+                {
+                    IUnconfiguredProjectVsServices _projectVsServices = UnconfiguredProject.Services.ExportProvider.GetExportedValue<IUnconfiguredProjectVsServices>();
+                    _projectThreadingService = _projectVsServices.ThreadingService;
+
+                }
+                return _projectThreadingService;
+            }
+        }
+
+        public event EventHandler ClearEnvironmentVariablesGridError;
+        public event EventHandler FocusEnvironmentVariablesGridRow;
+        public DebugPageViewModel()
+        {
+
+        }
+        // for unit testing
+        internal DebugPageViewModel(bool useTaskFactory,UnconfiguredProject unconfiguredProject)
+        {
+            _useTaskFactory = useTaskFactory;
+            UnconfiguredProject = unconfiguredProject;
+        }
+        private bool _useSpecificRuntime;
+        public bool UseSpecificRuntime
+        {
+            get
+            {
+                return _useSpecificRuntime;
+            }
+            set
+            {
+                OnPropertyChanged(ref _useSpecificRuntime, value);
+            }
+        }
+
+        public string SelectedCommandName
+        {
+            get
+            {
+                if (!IsProfileSelected)
+                {
+                    return string.Empty;
+                }
+
+                return SelectedDebugProfile.CommandName;
+            }
+            set
+            {
+                if (SelectedDebugProfile != null && SelectedDebugProfile.CommandName != value)
+                {
+                    SelectedDebugProfile.CommandName = value;
+                    OnPropertyChanged(nameof(SelectedCommandName));
+                }
+            }
+        }
+
+        private ObservableCollection<string> _commandNames;
+        public ObservableCollection<string> CommandNames
+        {
+            get
+            {
+                return _commandNames;
+            }
+            set
+            {
+                OnPropertyChanged(ref _commandNames, value);
+            }
+        }
+
+        /// <summary>
+        /// Where the data is bound depends on the currently active profile. IIS\IIS Express have specific binding information
+        /// in the IISSEttings. However, profiles (web based) have an ApplicationUrl property
+        /// </summary>
+        public string ApplicationUrl
+        {
+            get
+            {
+                if (SelectedDebugProfile != null)
+                {
+                    if (SelectedDebugProfile.Kind == ProfileKind.IISExpress)
```

Comment:
>SelectedDebugProfile.Kind == ProfileKind.IISExpress) [](start = 24, length = 52)

Change these to be SelectedDebugProfile.CommandName == "IISExpress". Don't reintroduce kind. This is the part which needs the extensibility. I might make more sense in the short term to literally comment out all the web related controls and code. Get that working, and then we can work on the extensibility for this UI